lib/TPP/ConvertPackOptimization.cpp

```diff
@@ -0,0 +1,202 @@
+////===ConvertPackOptimization.cpp ----------------------------*---- C++-*-===//
+////
+//// Part of the LLVM Project, under the Apache License v2.0 with LLVM
+/// Exceptions. / See https://llvm.org/LICENSE.txt for license information. /
+/// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+////
+////===----------------------------------------------------------------------===//
+//
+//
+
+#include "TPP/BuilderUtils.h"
+#include "TPP/Passes.h"
+#include "mlir/Dialect/Func/IR/FuncOps.h"
+#include "mlir/Dialect/SCF/IR/SCF.h"
+#include "mlir/Dialect/Tensor/IR/Tensor.h"
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+
+using namespace mlir;
+using namespace mlir::tpp;
+
+#define GEN_PASS_CLASSES
+#include "TPP/Passes.h.inc"
+
+namespace {
+
+struct MatchResult {
+  bool haveMatch;
+  int index;
+};
+
+/*
+ * Returns a list of indices that match the inner dimensions' position
```

Comment:
I still don't understand, sorry. The inner_dims_pos are already the indices that get tiled. You seems to return the same indices.

Reply:
I don't return whatever is in inner_dims_pos, but the tiling dimension for every inner_dims_pos index that matches the index of the tensor shape.